catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -109,48 +111,58 @@ public Metacard read(InputStream inputStream) throws CatalogTransformerException
                     getInputTransformerErrorHandler().configure(new StringBuilder());
             parser.setErrorHandler(inputTransformerErrorHandler);
             parser.parse(newStream);
+        } catch (IOException | SAXException e) {
+            throw new CatalogTransformerException("Could not properly parse metacard", e);
+        }
+
+        return this;
+    }
 
+    public Metacard getMetacard(String id) {
+        MetacardType metacardType = getMetacardType(id);
+
+        if (metacardType == null) {
```

Comment:
A debug level log statement inside this `if` branch saying that the metacard type was not dynamically found might be useful.